app/code/community/Bolt/Boltpay/Model/ShippingAndTax.php

```diff
@@ -202,12 +203,21 @@ public function getShippingAndTaxEstimate( $quote )
                 );
 
                 $response['shipping_options'][] = $option;
+
+                Mage::dispatchEvent(
+                    'bolt_boltpay_shipping_option_added',
```

Comment:
we already have top-level event for `bolt_boltpay_filter_shipping_and_tax_estimate`. do we need per-option event?

Reply:
In general, we are going to want to be very liberal about the inclusion of events in our system.  There have been several integrations that did not take into the long-term impact of the changes that were being made, and our goal is to provide for every opportunity possible for the integration engineer to not resort to bad design choices.

Events, as you mentioned, are a separate topic in its entirety and will hold back what is intended to be a quick and simple PR.   Instead of going file by file, which is some ways is easier and in some ways present some challenges in breaking up the code, I will go with the logical task division breaking down the code, which makes tracking progress and completion a bit more difficult, but should help us focus on one concept at a time.

Events are being removed for this PR and will be added to a future PR to speed up this process.

Reply:
thanks. btw I'm against adding event when we don't have concrete use case. If we add event it'll be public contract we have to maintain, let's only add when only needed, also we should be careful about what events are suitable to expose